♻️ Add back the device name with the running tab

AlexV525 opened the pull request:
Recovers the ability of https://github.com/flutter/flutter-intellij/pull/6774.
Fixes #7605
Fixes #7539 

`myDisplayName` no longer exists in the `com.intellij.execution.ui.RunContentDescriptor`, it has been replaced by a `MutableReactiveProperty` `myDisplayNameView` since 2024 builds.

[2024+ `myDisplayNameView`](https://github.com/JetBrains/intellij-community/blob/idea/231.8109.175/platform/execution/src/com/intellij/execution/ui/RunContentDescriptor.java#L30)
[2023 `myDisplayName`](https://github.com/JetBrains/intellij-community/blob/idea/241.14494.240/platform/execution/src/com/intellij/execution/ui/RunContentDescriptor.java#L33)

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

<!-- Links -->
[Contributor Guide]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md
[test-exempt]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/blob/master/docs/contributing/Style-guide-for-Flutter-repo.md
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/blob/master/docs/contributing/Chat.md

AlexV525 commented:
@jwren

pq reviewed (commented):
I'll defer to @jwren  or @helin24 who have more current context but as scary as this introspection is, it seems reasonable (if unfortunate). The guards seem to mitigate risk too.

pq commented on flutter-idea/src/io/flutter/run/LaunchState.java:
```diff
@@ -156,14 +158,31 @@ protected RunContentDescriptor launch(@NotNull ExecutionEnvironment env) throws
+    boolean displayNameUpdated = false;
     try {
-      final Field f = descriptor.getClass().getDeclaredField("myDisplayName");
+      final Field f = descriptor.getClass().getDeclaredField("myDisplayNameView");
```
Could you add a comment (somewhere) that explains why we have to do all this introspection? (And why we need to look for `"myDisplayNameView"` and fall back to `"myDisplayName"`? (Your PR description might be sufficient.)

AlexV525 replied:
Sure! Updated

pq replied:
Super. Thank you!

jwren commented:
Sorry for the delay.  The Flutter IJ plugin doesn't have tests for changes like this which is unfortunately.  @AlexV525, I have not had a chance to test / validate this change myself, but would be very happy to have it land.  If you have cycles, do add a few notes point out the history here in the code, and rebase to recent changes on `master`.  Regardless though of your time, this is a good change and I'll make sure it lands for either M85.1 or M85.2 if it works as you describe.

AlexV525 commented:
> If you have cycles, do add a few notes point out the history here in the code, and rebase to recent changes on `master`. 

I've added a few comments that indicate what the code is trying to do and why, also with src links to the InteliiJ Platform SDK.

The branch is also updated.
